### ISSUE TITLE
Question about the interface

### ISSUE THREAD

**Issue by 97-109-107** [junegunn/vim-journal#1]
Dear Junegunn,

I think I'm speaking for a lot of people here saying that I'm really looking forward for this to take shape :)

I see in your screenshot that you have a rich set of symbols in front of the lists and it got me curious about the interaction paradigm you're planning here. Looking and all those symbols, I was wondering if you will be relying on user-visible elements to change the attributes of a list element.

For example, assuming that you'd implement a task list:

`[ ] Buy milk`

There are at least two ways of thinking how to toggle it - ex. a key chord like 'cot', which in turn toggle the status of the task: 

`[X] Buy milk`

The other approach would be that just editing the `[ ]` element would change the status of it.

This is not a feature request, more like an open question as how would you see this coming together here? _The underlying motivations is that I would definitely like to try to contribute if the paradigm is aligned with my personal preference._

**Comment by junegunn** [comment]
Hi, it's too early to say anything. Currently it's just a personal settings for my note taking and I haven't thought about "releasing" it as a public plugin. It's on GitHub only because I find it easier to manage the code here.

I used [nvALT](http://brettterpstra.com/projects/nvalt/) with simplenote synchronization for the last few years for taking notes. But I recently decided to switch to Vim with Google Drive sync. And the purpose of this plugin is simply to make those plain-text notes I've written so far look nice on Vim. My note files of course were not written with some specific syntax/structure in mind, but still some common patterns can be found among them, such as indentation, bullet lists, etc, and this plugin is an attempt to extract and organize those patterns into a Vim syntax file.

The syntax file already highlights todo lists with check boxes, which is all I need right now, and I'm not particularly interested in extending this into an alternative to vim-notes, vim-orgmode or the likes.

![2015-04-27 11 15 54](https://cloud.githubusercontent.com/assets/700826/7349373/63675844-ed33-11e4-875f-d287e2d20c7d.png)
